proposals/0132-sequence-end-ops.md

```diff
@@ -119,31 +119,68 @@ These changes yield (bold parts are different):
+| ---------------------------------------------------------------- | ------------------ |
+| `Sequence`                                                       | `drop(while:)`     |
+| `Sequence`                                                       | `dropFirst()`      |
+| `Sequence`                                                       | `dropFirst(_:)`    |
+| `Sequence`                                                       | `dropLast()`       |
+| `Sequence`                                                       | `dropLast(_:)`     |
+| `Sequence`                                                       | `starts(with:by:)` |
+| `Sequence where Element: Equatable`                              | `starts(with:)`    |
+| `Collection`                                                     | `index(where:)`    |
+| `Collection where Element: Equatable`                            | `index(of:)`       |
+| `Collection where SubSequence == Self`                           | `removeFirst(_:)`  |
+| `RangeReplaceableCollection`                                     | `removeFirst(_:)`  |
+| `BidirectionalCollection where SubSequence == Self`              | `removeLast(_:)`   |
+| `RangeReplaceableCollection where Self: BidirectionalCollection` | `removeLast(_:)`   |
+
+(A couple more may be needed on `RangeReplaceableCollection` to resolve ambiguities with 
+`removeFirst(_:)` and `removeLast(_:)`.)
+
+Concrete implementations of the following methods, throughout the standard library, will 
+be renamed:
+
+| Current name                      | New name                                   |
+| --------------------------------- | ------------------------------------------ |
+| `index(of: Element)`              | `firstIndex(of: Element)`                  |
+| `index(where: (Element) -> Bool)` | `firstIndex(where: (Element) -> Bool)`     |
+| `dropFirst()`                     | `removingFirst()`                          |
+| `dropLast()`                      | `removingLast()`                           |
+| `dropFirst(_: Int)`               | `removingPrefix(_: Int)`                   |
+| `dropLast(_: Int)`                | `removingSuffix(_: Int)`                   |
+| `drop(while: (Element) -> Bool)`  | `removingPrefix(while: (Element) -> Bool)` |
+| `removeFirst(_: Int)`             | `removePrefix(_: Int)`                     |
+| `removeLast(_: Int)`              | `removeSuffix(_: Int)`                     |
+| `starts(with: PossiblePrefix)`    | `hasPrefix(_: PossiblePrefix)`             |
+| `starts(with: PossiblePrefix, by: (Element, Element) -> Bool)` | `hasPrefix(PossiblePrefix, by: (Element, Element) -> Bool)` |
+
+Some internal types, methods, and properties will be renamed as well.
+
+The following things are currently *not* being renamed, but perhaps should be:
+
+* `Collection._customIndexOfEquatableElement(_:)` requirement
+* `LazyDropWhileCollection` struct
```

Comment:
Why not rename these for consistency? Realistically, if they're not renamed now, they won't be in the future.

Granted, `LazyRemovingPrefixWhileCollection` is a mouthful, but we can revisit on the SE list whether `drop(while:)` needs to be renamed to `removingPrefix(while:)` or whether "while" is sufficiently evocative that it can be simply `removing(while:)`, and thus `LazyRemovingWhileCollection`.

Reply:
Honestly, the main reason I haven't renamed `LazyDropWhileCollection` is that I didn't want to deal with renaming the file it's in, DropWhile.swift.gyb.

I left `_customIndexOfEquatableElement(_:)` alone because it's already internal, it's already out of date (it should be `_customIndex(ofEquatableElement:)` or something), I suspect there are custom implementations out there even though it's internal and don't know how to make the compiler reject them, and I'm sad we can't make the requirement public as `func firstIndex(of: Element) -> Index? where Element: Equatable`.

Bottom line is, I wasn't sure what to do about these and I was hoping someone would have good ideas.

(I'll leave the `removing(while:)` discussion for the review.)